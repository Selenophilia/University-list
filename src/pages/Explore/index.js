import React, { useEffect, useState } from "react";
import { makeStyles } from "@material-ui/core/styles";
import { Typography } from "@material-ui/core";
import { fetchUniversities } from "../../api";
import InfiniteScroll from "react-infinite-scroll-component";
import { UniversityCard } from "../../components";
import Search from "../../components/Search";

const useStyles = makeStyles(() => ({
  explore: {
    paddingLeft: 100,
    paddingRight: 100,
    marginBottom: 100,
    display: "flex",
    flexDirection: "column"
  },
  title: {
    fontSize: 32,
    fontWeight: "bold",
    marginBottom: 20
  },
  universities: {
    display: "flex",
    flexWrap: "wrap",
    alignContent: "center",
    justifyContent: "space-between"
  },
  filters: {
    display: "flex",
    marginBottom: 50
  }
}));

const Explore = () => {
  const classes = useStyles();
  const [university, setUniversity] = useState(null);
  const [count, setCount] = useState({
    prev: 0,
    next: 10
  });
  const [hasMore, setHasMore] = useState(true);
  const [current, setCurrent] = useState("");
  const handleData = async () => {
    const output = await fetchUniversities();

    const data = output.map((uni, idx) => ({
      id: idx,
      name: uni.name,
      country: uni.country,
      page: uni.web_pages[0]
    }));
    setUniversity(data);
    setCurrent(data.slice(count.prev, count.next));
  };

  const getMoreData = () => {
    if (current.length === university.length) {
      setHasMore(false);
      return;
    }
    setTimeout(() => {
      setCurrent(
        current.concat(university.slice(count.prev + 10, count.next + 10))
      );
    }, 2000);
    setCount((prevState) => ({
      prev: prevState.prev + 10,
      next: prevState.next + 10
    }));
  };
  useEffect(() => {
    handleData();
  }, []);
  return (
    <div className={classes.explore}>
      <div className={classes.title}>
        <Typography variant="h3" className={classes.title}>
          Explore
        </Typography>
      </div>
      <div className={classes.filters}>
        <Search />
      </div>
      {university && (
        <InfiniteScroll
          dataLength={current.length}
          next={getMoreData}
          hasMore={hasMore}
          loader={<h4>Loading...</h4>}
          className={classes.universities}
        >
          <div className={classes.universities}>
            {current &&
              current.map((uni, index) => (
                <UniversityCard
                  key={index}
                  image={
                    "https://images.unsplash.com/20/cambridge.JPG?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=730&q=80"
                  }
                  name={uni.name}
                  country={uni.country}
                  page={uni.page}
                />
              ))}
          </div>
        </InfiniteScroll>
      )}
    </div>
  );
};

export default Explore;
