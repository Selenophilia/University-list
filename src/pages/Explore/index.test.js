import React from "react";
import { render } from "@testing-library/react";
import Explore from ".";

const renderRoot = () => {
  return render(<Explore />);
};

it("renders Explore component correctly ", () => {
  const { asFragment } = renderRoot();
  expect(asFragment()).toMatchSnapshot();
});
