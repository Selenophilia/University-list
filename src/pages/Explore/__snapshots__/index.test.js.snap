// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Explore component correctly  1`] = `
<DocumentFragment>
  <div
    class="makeStyles-explore-1"
  >
    <div
      class="makeStyles-title-2"
    >
      <h3
        class="MuiTypography-root makeStyles-title-2 MuiTypography-h3"
      >
        Explore
      </h3>
    </div>
    <div
      class="makeStyles-filters-4"
    >
      <form
        class="makeStyles-searchForm-5"
      >
        <input
          class="makeStyles-field-6"
          name="ukuusearch"
          placeholder="Search item"
          type="text"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root makeStyles-icon-7"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
      </form>
    </div>
    <div
      class="infinite-scroll-component__outerdiv"
    >
      <div
        class="infinite-scroll-component makeStyles-universities-3"
        style="height: auto; overflow: auto;"
      >
        <div
          class="makeStyles-universities-3"
        />
        <div
          class="container"
          data-testid="loader"
        >
          <i
            class="loading"
            type="icon-spinner animate-spin"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
